docs/functions/building-functions.md

```diff
@@ -0,0 +1,42 @@
+# Building functions
+
+--8<-- "build-func-intro.md"
+
+## Local builds
+
+You can build a container image for your function locally without deploying it to a cluster, by using the `build` command.
+
+### Prerequisites
+
+- You have a Docker daemon on your local machine. This is already provided if you have used the Quickstart installation.
+
+### Procedure
+
+--8<-- "proc-building-function.md"
+
+## On-cluster Builds
+
+If you do not have a local Docker daemon running, or you are using a CI/CD pipeline, you might want to build your function on the cluster instead of using a local build. You can create an on-cluster build by using the `func deploy --remote` command.
+
+### Prerequisites
+
+- The function must exist in a Git repository.
```

Comment:
This is not entirely true for the current release. We have just added the ability to do direct source upload so that you don't have to have your function in a git repo. The procedure is almost identical, but instead of the cluster pulling function source code from github, the source is pushed to the cluster directly from the developer's laptop. This is the behavior when issuing the command `func deploy --remote` with no `--git-url` flag provided.

Reply:
By current release do you mean 1.7 or 1.8?

Reply:
1.8

Reply:
In that case I'd add it as an additional PR so that this one can be CP'd back to older versions